subscription.go

```diff
@@ -162,19 +162,21 @@ func (r *marathonClient) registerCallbackSubscription() error {
 	return nil
 }
 
-func (r *marathonClient) registerSSESubscription() error {
-	// Prevent multiple SSE subscriptions
+func (r *marathonClient) registerSSESubscriptionOnce() error {
 	if r.subscribedToSSE {
 		return nil
 	}
 
-	request, _, err := r.buildAPIRequest("GET", marathonAPIEventStream, nil)
-	if err != nil {
+	if err := r.registerSSESubscription(); err != nil {
 		return err
 	}
 
-	// Try to connect to stream, reusing the http client settings
-	stream, err := eventsource.SubscribeWith("", r.config.HTTPClient, request)
+	r.subscribedToSSE = true
+	return nil
+}
+
+func (r *marathonClient) registerSSESubscription() error {
```

Comment:
Okay on the part that this should be handled by a different PR as I understood the method didn't "grew worse" in this regards. 👌 

AFAICS though, it's more than a nice-to-have since we seem to be leaking a goroutine per subscription. I'm thinking of applications that use go-marathon to continuously subscribe and unsubscribe to the SSE stream without ever terminating. Or did I misunderstood?